src/main/scala/cognite/spark/v1/fdm/FlexibleDataModelBaseRelation.scala

```diff
@@ -583,7 +585,13 @@ abstract class FlexibleDataModelBaseRelation(config: RelationConfig, sqlContext:
     }
 }
 
-object FlexibleDataModelBaseRelation extends NamedRelation {
+object FlexibleDataModelBaseRelation
+    extends NamedRelation
+    with UpsertSchema
+    with DeleteWithExternalIdSchema {
```

Comment:
might well be a bug but schema in https://github.com/cognitedata/jetfire-backend/blob/master/driver/src/main/scala/com/cognite/jetfire/driver/SchemaService.scala#L102 is full object, also per-fdm-type rather than shared via base. And FlexibleDataModelConnectionRelation is using something wider than just external id too

Reply:
Aren't they all using the exact same predefined type?

Reply:
by the way this predefined type is actually   val dataModelInstanceSchema = StructType(Seq(StructField("externalId", DataTypes.StringType))) not full object, or am I missing something here?.

Reply:
Hmm, got blinded by `.dataModelInstanceSchema` a bit, so that indeed remains, but FlexibleDataModelConnectionRelation.delete goes via FlexibleDataModelRelationUtils.createEdgeDeleteData and also takes spaceId if available in row. So I guess it's then a bug in exposed schema :thisisfine: that we can look at in a separate PR :+1: